Geometry/HcalCommonData/interface/HcalDDDRecConstants.h

```diff
@@ -80,29 +80,18 @@ class HcalDDDRecConstants {
        
 private:
   void                      initialize(void);
+  unsigned int              layerGroupSize( unsigned int eta ) const;
+  unsigned int              layerGroup( unsigned int eta, unsigned int i ) const;
 
-  bool                tobeInitialized;
-  static const int    nEtaMax=100;
   const HcalParameters      *hpar;
   const HcalDDDSimConstants &hcons;
-  int                 modeTopo_;  // Mode for topology
-  std::vector<int>    etaGroup;   // Eta Grouping
   std::vector<std::pair<int,int> > etaSimValu; // eta ranges at Sim stage
-  std::vector<double> etaTable;   // Eta table (HB+HE)
   std::vector<int>    ietaMap;    // Map Sim level ieta to Rec level ieta
-  std::vector<int>    iEtaMin, iEtaMax; // Minimum and maximum eta
-  std::vector<int>    maxDepth;   // Maximum depth in HB/HE/HF/HO           
-  int                 nEta;       // Number of bins in eta for HB and HE
-  std::vector<int>    phiGroup;   // Eta Grouping
   std::vector<double> phibin;     // Phi step for all eta bins (HB, HE, HO)
```

Comment:
I dont know which could be persistent and which could be transient variables. This depends on the analysis if you could make some computation at the beginning and use them later on. At this stage I suggest please do not change the payload and whatever can be calculated may be there at initialization.
________________________________
From: Ianna Osborne [notifications@github.com]
Sent: 15 July 2015 16:50
To: cms-sw/cmssw
Cc: Sunanda Banerjee
Subject: Re: [cmssw] Hcal DD Rec and Sim Constants Cleanup (#10214)


In Geometry/HcalCommonData/interface/HcalDDDRecConstants.h<https://github.com/cms-sw/cmssw/pull/10214#discussion_r34686162>:

>    std::vector<int>    ietaMap;    // Map Sim level ieta to Rec level ieta
> -  std::vector<int>    iEtaMin, iEtaMax; // Minimum and maximum eta
> -  std::vector<int>    maxDepth;   // Maximum depth in HB/HE/HF/HO
> -  int                 nEta;       // Number of bins in eta for HB and HE
> -  std::vector<int>    phiGroup;   // Eta Grouping
>    std::vector<double> phibin;     // Phi step for all eta bins (HB, HE, HO)


@bsunanda<https://github.com/bsunanda> - would it make sense to add the remaining vectors which are calculated at initialisation of this class to HcalParameters? The same for Sim part?

—
Reply to this email directly or view it on GitHub<https://github.com/cms-sw/cmssw/pull/10214/files#r34686162>.